compiler/AST/AggregateType.cpp

```diff
@@ -2209,7 +2223,7 @@ void AggregateType::processGenericFields() {
       if (field->defPoint->exprType == NULL) {
         genericFields.push_back(field); // "var x;"
         isGenericWithDefaults = false;
-      } else if (isFieldTypeExprGeneric(field->defPoint->exprType)) {
+      } else if (isFieldTypeExprGeneric(field->defPoint->exprType, visited)) {
```

Comment:
I am not a big fan of recursion's side effects (`visited`) being declared in this scope and wish we could have a helper for here, as well. But that's too much about personal taste and not a significant issue :)